src/amplitude/event.py

```diff
@@ -239,6 +240,7 @@ def __init__(self, user_id: Optional[str] = None,
         self.session_id: Optional[int] = None
         self.insert_id: Optional[str] = None
         self.library: Optional[str] = None
+        self.library_context: Optional[str] = None
```

Comment:
Is the library_context designed to not be part of the constructor parameters?

Reply:
Correct me if I am wrong, I think Python SDK is different, it has `EventOptions`, I didn't find similar things in other SDKs. In my understanding, it is a one-time overwrite to provide additional information to the event. If this is the case, I think we probably don't need this in the constructor, as the `library_context` should be always a global information. What is you suggestion on this?

Reply:
EventOptions are introduced in our new gen SDK e.g. [ts](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-types/src/base-event.ts) [kotlin](https://github.com/amplitude/Amplitude-Kotlin/blob/main/core/src/main/java/com/amplitude/core/events/EventOptions.kt). It's designed for ampli that have all attributes except those controlled by tracking plan. Usually used to provide additional info to events but limit the ability to override ampli controlled part. It's also base class of other event types classes. There are some fields that we used internally like library and plan. Maybe it's ok to not include them in the constructor. I think adding library_context as optional parameter for consistency.

Reply:
Got it, thanks for the explanation! Then I will keep it excluded from the constructor like what we have here now.